misoc/integration/soc_core.py

```diff
@@ -135,9 +135,9 @@ def register_mem(self, name, origin, length, interface):
 
     def register_rom(self, interface, rom_size=0xa000):
         self.add_wb_slave(self.mem_map["rom"], rom_size, interface)
-        assert self.cpu_reset_address < rom_size
-        self.add_memory_region("rom", self.cpu_reset_address,
-                               rom_size-self.cpu_reset_address)
+        if not self.mem_map["rom"] <= self.cpu_reset_address < self.mem_map["rom"] + rom_size:
```

Comment:
Ok.

Reply:
agreed that self.cpu_reset_address == self.mem_map["rom"]  hence changing this code